text/0115-circuit-breaker.md

```diff
@@ -0,0 +1,110 @@
+# Title
+
+- RFC PR: https://github.com/tikv/rfcs/pull/115
+- Tracking Issue: https://github.com/tikv/pd/issues/8678
+
+## Summary
+
+Implement a [circuit breaker](https://learn.microsoft.com/en-us/azure/architecture/patterns/circuit-breaker) pattern to protect the PD leader from overloading due to retry storms or similar feedback loops.
+
+## Motivation
+
+In large TiDB clusters with hundreds of tidb and tikv nodes, the PD leader can become overwhelmed during certain failure conditions, leading to a "retry storm" or other feedback loop scenarios. 
+Once triggered, PD transitions into a metastable state and cannot recover autonomously, leaving the cluster in a degraded or unavailable state.
+This feature is especially critical for large clusters where a high number of nodes can continuously hammer a single PD instance during failures, leading to cascading effects that worsen recovery times and overall cluster health.
+
+## Detailed design
+
+### Configuration
+
+The key part of the circuit breaker is its configuration. 
+In order to limit number of system variables, we will use a single system variables per circuit breaker type to define error rate threshold. 
+
+#### System variables
+
+`tidb_cb_pd_metadata_error_rate_threshold_pct`:
```

Comment:
Seems
1. `pd_metadata` is difficult to understand
2. `error_rate` is redundant with pct

How about renaming it to `pd_client_cb_error_rate_threshold`?

Reply:
If I remember correctly, the conclusion of the last meeting was that a single switch should control all interfaces.

Reply:
Do you have any suggestions about this? @rleungx @nolouch

Reply:
If we have a dedicated circuit breaker for get regions calls, do we need some variables to control other calls?

Reply:
Do you mean we are only developing the region metadata part this time?🤔 @Tema

Reply:
IMO, it is better to manage an interface with a variable. For the circuit breaker itself, different interfaces can share the same implementation and only use a different setting. Right now we can focus on the GetRegion call. /cc @okJiang